pkg/ck8s/workload_cluster.go

```diff
@@ -300,6 +355,58 @@ func (w *Workload) doK8sdRequest(ctx context.Context, method, endpoint string, r
 	return nil
 }
 
+func (w *Workload) doK8sdRequestForMachine(ctx context.Context, machine *clusterv1.Machine, method, endpoint string, header map[string][]string, request any, response any) error {
+	type wrappedResponse struct {
+		Error    string          `json:"error"`
+		Metadata json.RawMessage `json:"metadata"`
+	}
+
+	k8sdProxy, err := w.GetK8sdProxyForMachine(ctx, machine)
+	if err != nil {
+		return fmt.Errorf("failed to create k8sd proxy: %w", err)
+	}
+
+	url := fmt.Sprintf("https://%s:%v/%s", k8sdProxy.NodeIP, w.microclusterPort, endpoint)
+
+	requestBody, err := json.Marshal(request)
+	if err != nil {
+		return fmt.Errorf("failed to prepare worker info request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header = http.Header(header)
+
+	res, err := k8sdProxy.Client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to reach k8sd through proxy client: %w", err)
+	}
+	defer res.Body.Close()
+
+	var responseBody wrappedResponse
+	if err := json.NewDecoder(res.Body).Decode(&responseBody); err != nil {
+		return fmt.Errorf("failed to parse HTTP response: %w", err)
+	}
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("HTTP request failed with status code: %d (%s)", res.StatusCode, responseBody.Error)
+	}
+	if responseBody.Error != "" {
+		return fmt.Errorf("k8sd request failed: %s", responseBody.Error)
+	}
+	if responseBody.Metadata == nil {
```

Comment:
let's also add this, otherwise we might encounter an error on `json.Unmarshal` if a nil response is passed (we had the same problem in `doK8sdRequest` for controlplane nodes).
```suggestion
	if responseBody.Metadata == nil || response == nil {
```